From 7e8dff53b7fcf4b6ac92413531affdb40e9c7f12 Mon Sep 17 00:00:00 2001
From: Christina Jacob <cjacob@marvell.com>
Date: Wed, 15 May 2019 11:21:32 +0530
Subject: [PATCH 0184/1921] octeontx2-af: Move to rvu_fwdata version 1.

- Update rvu_fwdata to match with latest definition in atf.
- The latest changes ensures that the offset of already existing
  member variables does not change on adding new variables.
  Thus the existing variables does not get affected evenif
  the definitions in atf and kernel doesn't match exactly.
- Additions to cgx_lmac_fwdata_s is restricted to the end
  of the structure.
- Add sff_id(small form factor) to the sfp_eeprom_s structure.
  Required to specify the module type while reading transceiver
  eeprom information via ethtool

Change-Id: I5702955c2ca37ef29c41c80edd5955694619f410
Signed-off-by: Christina Jacob <cjacob@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/9072
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/mbox.h | 2 +-
 drivers/net/ethernet/marvell/octeontx2/af/rvu.h  | 8 +++++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
index 07d09b4efd06..3a66919309d3 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
@@ -469,7 +469,7 @@ struct cgx_pause_frm_cfg {
 
 struct sfp_eeprom_s {
 #define SFP_EEPROM_SIZE 256
-	u16 valid;	/* to be set to 1 after copying data to buf */
+	u16 sff_id;
 	u8 buf[SFP_EEPROM_SIZE];
 };
 
diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu.h b/drivers/net/ethernet/marvell/octeontx2/af/rvu.h
index 19218e2922ee..acb7c6ad6456 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu.h
@@ -372,7 +372,7 @@ struct mbox_wq_info {
 
 struct rvu_fwdata {
 #define RVU_FWDATA_HEADER_MAGIC	0xCFDA	/*Custom Firmware Data*/
-#define RVU_FWDATA_VERSION	0x0000
+#define RVU_FWDATA_VERSION	0x0001
 	u32 header_magic;
 	u32 version;		/* version id */
 
@@ -381,6 +381,12 @@ struct rvu_fwdata {
 #define VF_MACNUM_MAX	256
 	u64 pf_macs[PF_MACNUM_MAX];
 	u64 vf_macs[VF_MACNUM_MAX];
+	u64 sclk;
+	u64 rclk;
+	u64 mcam_addr;
+	u64 mcam_sz;
+#define RESERVED_MEM 1024
+	u64 reserved[RESERVED_MEM];
 #define CGX_MAX         3
 #define CGX_LMACS_MAX   4
 	struct cgx_lmac_fwdata_s cgx_fw_data[CGX_MAX][CGX_LMACS_MAX];
-- 
2.31.1

